hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryWorkflowImpl.java

```diff
@@ -302,15 +312,15 @@ public void handleQuery(@NonNull final Bytes requestBuffer, @NonNull final Buffe
         }
     }
 
-    private Query parseQuery(Bytes requestBuffer) {
+    private Query parseQuery(Bytes requestBuffer) throws PreCheckException {
         try {
             return queryParser.parseStrict(requestBuffer.toReadableSequentialData());
         } catch (ParseException e) {
             switch (e.getCause()) {
                 case MalformedProtobufException ex:
                     break;
                 case UnknownFieldException ex:
-                    break;
+                    throw new PreCheckException(TRANSACTION_HAS_UNKNOWN_FIELDS);
```

Comment:
I think we cannot send a proper response code if we cannot parse the query.

Reply:
We can parse the query. UnknownFieldException means that we know what type of query was sent, but there is an extra or misspelled field in the request.

Reply:
For transactions, see TransactionChecker:486 for how similar errors in transactions are dealt with

Reply:
You are right. We could probably determine the type of query if we encounter an `UnknownFieldException`. However, in strict mode, PBJ does not provide this information but fails. We could try to parse the query a second time in non-strict mode to get the type. But then again, it is a user error, and nobody pays the node operator, so why bother? 🙂

Transactions are fundamentally different. Every transaction expects a serialized `Response`. Even if we do not know the transaction type, we can return a `Response`. However, in the case of queries, every query has a different response type. If a client sends a `CryptoGetInfoQuery`, it expects a serialized `CryptoGetInfoResponse`. If we return a `TransactionGetReceiptResponse`, it will just see a byte array that it cannot interpret. Due to this design choice, we can only return a proper response once we know the query type.

Reply:
See my response above.